charts/core/src/Line/Line.tsx

```diff
@@ -6,14 +6,22 @@ import { defaultLineOptions } from './config';
 import { LineProps } from './Line.types';
 
 export function Line({ name, data }: LineProps) {
-  const { addChartSeries } = useChartContext();
+  const { addChartSeries, removeChartSeries } = useChartContext();
 
   useEffect(() => {
     addChartSeries({
       ...defaultLineOptions,
       name,
       data,
     });
+
+    return () => {
+      /**
+       * Remove the series when the component unmounts to make sure the series
+       * is removed when a `Line` is hidden.
+       */
+      removeChartSeries(name);
```

Comment:
The only place I see a potential issue is if the consumer has rendered another line with the same name before this one unmounts, and we then remove the wrong line from the chart. But at that point, I'd almost call that "user error"

Reply:
Probably a check within `addChartSeries` that logs something to the console, and exits early?

Reply:
Yeah and actually `addChartSeries` is buggy atm. It adds duplicates on re-renders. Was going to fix that in a followup but maybe I should just fix that here. Will push up this fix shortly

Reply:
So oddly I'm finding that `addChartSeries` ends up getting called twice on render, even if the `useEffect` in `Line` only runs once. I'm kind of confused as to why 🤔. That said, this means that even legit series are sometimes getting added twice. I've added logic to prevent duplication and updated the docs to say that it needs to be unique and that non-uniquely named series won't be rendered, however I didn't add an error log due to it triggering on legit cases. Do you think that's enough?